### PR TITLE
Initial API

### PR DESCRIPTION
**Pull request by tig567899**
Returns player data if the user is authenticated in the website.

### CODE REVIEW

**Review by firemaster1294 — COMMENTED** [review]
I would like @pkrshnan to read over these changes before merging

**Comment by tig567899** [comment]
> 
> 
> I would like @pkrshnan to read over these changes before merging

Yep that was the intent - no rush - the client side isn't close to finished

**Comment by pkrshnan** [comment]
Couple big issues I see with this PR that I'd like to see changed before it goes further:
1. The API login page shouldn't be able to be viewed through a browser, i.e it should be returning a 204 No Content, not an HTTP 200. 
2. There should be docs written somewhere about how to test this API, maybe even in this PR.
3. This branch needs to be rebased off master, and should be in a separate branch entirely.

These are in addition to all the other comments on the PR. Looks good so far!

File: app/api/views/api_view.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from app.models import Player, Spectator, Moderator
+
+from django.http import Http404, JsonResponse
+
+import json
+
+def account_info(request):
+    if not request.user.is_authenticated:
+        print("Not Authenticated")
+        raise Http404
```
Comment (pkrshnan) [inline app/api/views/api_view.py]: This actually shoudn't be a 404. This should maybe be a 403 Unauthorized instead.

Comment (pkrshnan) [inline app/api/views/api_view.py:11]: This actually shoudn't be a 404. This should maybe be a 403 Unauthorized instead.

File: app/migrations/0032_auto_20200131_0606.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 2.2.9 on 2020-01-31 06:06
+
+from django.db import migrations, models
```
Comment (pkrshnan) [inline app/migrations/0032_auto_20200131_0606.py:3]: I'm a bit confused on why this migration exists. This shouldn't be part of this PR.

File: app/serializers/player_serializer.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        fields = ['code', 
+            'roleChar', 
+            'is_oz',
+            'name',
+            'email',
+            'score',
+            'shop_score',
+            'active',
+            'game',
+            'faction']
```
Comment (pkrshnan) [inline app/serializers/player_serializer.py:39]: We shouldn't have fields like `active` here, as we should only be returning active players anyway.

File: uwhvz/settings/common.py [hunk 4]
```diff
@@ -157,6 +157,11 @@
     messages.ERROR: 'danger',
 }
 
+REST_FRAMEWORK = {
+    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.PageNumberPagination',
+    'PAGE_SIZE': 10
```
Comment (pkrshnan) [inline uwhvz/settings/common.py:162]: Doesn't this seem small for a `PAGE_SIZE`? Considering we can be returning things like a player list of above a 100 players, aggregating across 10 pages seems more work for no reason. Maybe increase this to something like 50?

File: app/api/views/api_view.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        raise Http404 
+
+    if request.method == "GET":
+        playerSet = Player.objects.all().filter(user=request.user)
+        if playerSet:
+            player = playerSet[0]
+            serializer = PlayerSerializer(player)
+            return JsonResponse(serializer.data, safe=False)
+        else:
+            print("Not found")
```
Comment (pkrshnan) [inline app/api/views/api_view.py:20]: I'm not sure if we should be printing here, or if we are, it should give a lot more context as to what was not found and what the issue is. The same holds for above.

**Review by pkrshnan — CHANGES_REQUESTED** [review]
Good start, I'd like to see a bunch of cleanup around authorization before it goes any further!

File: uwhvz/settings/common.py (hunk 4 above)
Comment (tig567899) [reply uwhvz/settings/common.py:162]: We're not though - we're returning one set of data for one player - 10 should be more than enough. We can increase it if we expand the API to return moderator stuff.

File: app/serializers/player_serializer.py (hunk 3 above)
Comment (tig567899) [reply app/serializers/player_serializer.py:39]: That's fair.

**Comment by tig567899** [comment]
Closing this PR in favour of the new branch I opened. Will update API documentation.
